src/consensus/dispatch.ml

```diff
@@ -0,0 +1,78 @@
+open Steps
+
+let rec dispatch effect step state =
```

Comment:
I think this function could be cleaned up by capturing effect in a closure and removing it from the function parameters (because we shouldn't be modifying the effect handler, right?). E.g.:
```ocaml
let dispatch effect step state = 
  let rec dispatch step state = 
  and check_block ~block state = ...
  and append_blcok ~block state = ...
  ...
```

Reply:
I also think an MLI would help. I know, it's just one function but still.

Reply:
Then you add another context and another level of indentation, the flat version is just simpler, you're "cleaning it up" and also making each one of the functions harder to analyse as they're now closures.